tools/corpus/fetch_sources.py

```
@@ -0,0 +1,123 @@
+    url = 'https://zh.wikisource.org/w/index.php?' + urllib.parse.urlencode(
+        {'action': 'raw', 'title': title})
+    req = urllib.request.Request(url, headers={'User-Agent': UA})
+    for attempt in range(3):
+        try:
+            with urllib.request.urlopen(req, timeout=30) as r:
+                body = r.read()
+            if not body or body.startswith(b'#REDIRECT'):      # 리다이렉트 문서는 본문이 아니다.
+                return None
+            with open(path, 'wb') as f:
+                f.write(body)
+            return name
```

Ruff (0.16.1)

[warning] Consider moving this statement to an `else` block

(TRY300)